stream-chat-android-client/src/main/java/io/getstream/chat/android/client/parser2/MoshiChatParser.kt

```diff
@@ -59,6 +61,8 @@ internal class MoshiChatParser : ChatParser {
     private val moshi: Moshi by lazy {
         Moshi.Builder()
             .addAdapter(DateAdapter())
+            .addAdapter(ExactDateAdapter())
+            .addAdapter(ExactDateUpstreamAdapter())
```

Comment:
This adapter shouldn't be needed if we remove `ExactDateUpstream` class

Reply:
We need it for downstream. In order to create the `SyncState` with both `lastSyncedAt` and `rawLastSyncedAt` we need a different adapter that covers both `Date` and `String` otherwise we would be forced to choose between one of then.